frontend/src/pages/[username]/[projectName]/index.js

```diff
@@ -0,0 +1,413 @@
+    setIsSyncing(status === 'Queue' || status === 'Running')
+
+    if (props.isEmpty && !props.isSynced && status === 'Finished') {
+      reload()
+    }
+  }, [status])
+
+  if (isLoading) {
+    return (
+      <Page>
+        <Loader active />
+      </Page>
+    )
+  } else if (isSyncing) {
+    return (
+      <Page>
+        <Loader active>Syncing repository...</Loader>
+      </Page>
+    )
+  } else if (status === 'Failed') {
+    return (
+      <Page>
+        <Loader active>Migration Failed, please try again later!</Loader>
+      </Page>
+    )
+  } else if (isError) {
+    return <ErrorPage statusCode={404} />
+  }
+
+  return (
+    <Page>
+      {props.isSynced && props.hasUploadPermission ? (
+        <Message data-cy="sync-msg" color="yellow">
+          <Message.Header>A synced repository!</Message.Header>
+          <Message.Content>
+            <p>Files uploading isn't supported for synced repositories.</p>
+            Please commit files to the original git repository and it will be synced
+            automatically.
+          </Message.Content>
+        </Message>
+      ) : null}
+      <UpdateForm
+        {...props}
+        description={project?.description}
+        previewOnly={props.isSynced}
+        url={project?.original_url}
+        owner={props.user}
+        name={props.projectName}
+      />
+    </Page>
+  )
+}
+
+const UpdateForm = ({
+  repoFiles,
+  hasUploadPermission,
+  hasIBOM,
+  kitspaceYAML,
+  boardInfo,
+  zipUrl,
+  renderedReadme,
+  boardSpecs,
+  isNew,
+  previewOnly,
+  owner,
+  name,
+  description,
+  url,
+  boardAssetsExist,
+  readmeExists,
+  kitspaceYAMLExists,
+}) => {
+  const projectFullname = `${owner}/${name}`
+  // The files in the Gitea repo associated with this project and the newly loaded files
+  const {
+    files: remoteFiles,
+    isLoading,
+    mutate,
+  } = useDefaultBranchFiles(projectFullname, { initialData: repoFiles })
+  // Details(name, path, last_modified, etc...) for files dropped in the update page
+  const [newlyUploadedDetails, setNewlyUploadedDetails] = useState([])
+  const [allFiles, setAllFiles] = useState([])
+  const [isValidProjectName, setIsValidProjectName] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const { push } = useRouter()
+  const { csrf, user } = useContext(AuthContext)
+  const { form, onChange, populate, isValid, formatErrorPrompt } = useForm(
+    ProjectUpdateFormModel,
+  )
+  const [canUpload, setCanUpload] = useState(hasUploadPermission && !previewOnly)
+
+  // Set values of the form as the values of the project stored in the Gitea repo
+  useEffect(() => {
+    populate({ name, description }, true)
+  }, [])
+
+  useEffect(() => {
+    // Handle client side rendering for uploading permissions,
+    // `canUpload` previously relied on `hasUploadPermission` which is only provided in SSR mode.
+    if (!hasUploadPermission) {
+      setCanUpload(user?.username === owner && !previewOnly)
```

Comment:
Here again, needs to check Gitea API if user has write permission.

Reply:
Resolved via e998119.